mappings/clacso.py

```diff
@@ -1,23 +1,102 @@
-from .json import JSONMapping
+import re
+import requests
+from bs4 import BeautifulSoup
 
-class CLACSOMapping(JSONMapping):
-    def __init__(self, source):
-        super().__init__(source, {})
+from mappings.xml import XMLMapping
+
+class CLACSOMapping(XMLMapping):
+
+    HANDLE_URL = "https://biblioteca-repositorio.clacso.edu.ar/handle/CLACSO"
+
+    
+    def __init__(self, source, namespace, constants):
+        super(CLACSOMapping, self).__init__(source, namespace, constants)
         self.mapping = self.createMapping()
     
     def createMapping(self):
```

Comment:
I think the createMapping function hinders us quite a bit and is a roundabout way to map from A->B. What we are actually doing is going from A->C via the createMapping and applying the formatting in the XML mapping to get the values that we can from the XML source document. We then apply the formatting below to go from C->B.  

I think we can go directly from A->B. It will be more concise and easier to read. That being said, I'm open to pushback here. What are the benefits to keeping the createMapping function and this hierarchy of inherited mapping classes?

Reply:
From our conversation earlier, I agree with the shift to a more functional programming lens.